### ISSUE TITLE
Provide way to force a value to be treated as a ScalarValue in ILogger.ForContext

### ISSUE THREAD

**Issue by bartelink** [serilog/serilog#1124]
TL;DR Question: is there a way for me to tell `ILogger.ForContext` to not apply any transformations of any kind to the value I'm supplying, forcing it to be a `ScalarValue` ?
Suggestion/feature request based on ramble below: Have the OOTB config of Serilog not transform ForContext properties that are wrapped in `Serilog.Events.ScalarValue`

I'm on Serilog 2.6 (need to support 452 and Core, but platform is not really relevant) and am running into barriers / confusion wrt destructuring of properties that are not used in the message template (other than via `{Properties}`). (I happen to be consuming via `Serilog.Sinks.Observable`, but assume that's not a factor.)

I have no exotic config settings - I'm working in a low dependency library and am attempting to Just Work in whatever config is in place given an `ILogger` passed into the library's API. As I traverse layers therein, I enrich the context using `ILogger.ForContext(string,object)`.

For many messages, my final step before the `.Information` is to feed in a F# Discriminated union with captured metrics. The idea is that a `LogEvent` consumer can then pick out the Event data and then
- log it (I do a custom rendering which could be implemented in a `.ToString()` on the event and could probably be described by a Serilog destructuring scheme)
- feed the lower level info into a metrics repo (Prometheus) by concocting a set of properties derived from the Event 

In my initial impl, this data was an F# Record, and `Destructure.AsScalar()` allowed me to inhibit the destructuring.

Now I'm making that a DU, but that's running me into trouble - [as you know, excuse me explaining stuff you likely know intimately, but I feel it's important for me to surface my assumptions]:
- I assume I could make it work (but have not tried) by doing a `Destructure.AsScalar` for each concrete type in the union (don't want to do that as it would be messy and inefficient)
- For the longest time it confused me that clearly some destructuring controls where operating (actually only `Destructure.AsScalar`, but the fact that it lives under `.Destructure` led me to assume the others work in similar fashion) but were not being hit
- Eventually, I hit on the `destructureObjects` param on ILogger.ForContext and said "Eureka, `false`, done!"; but that's the default anyway
- Finally, I have made it work by explicitly requesting destructuring, and then neutralizing it via config

Code:
```
module Log =
    [<NoEquality; NoComparison>]
    type ReqMetric = { stream: string; interval: StopwatchInterval; bytes: int; count: int }
    [<NoEquality; NoComparison>]
    type Evt =
        | WriteSuccess of ReqMetric
        | WriteConflict of ReqMetric
        | Slice of Direction * ReqMetric
        | Batch of Direction * slices: int * ReqMetric
    /// Inhibit the destructuring of a type or any derived types
    type DontDestructureAssignableTo<'T>() =
        let isaT (value : obj) = typedefof<'T>.IsAssignableFrom(value.GetType())

        interface Serilog.Core.IDestructuringPolicy with
            member __.TryDestructure(value, _propertyValueFactory, result: byref<Serilog.Events.LogEventPropertyValue>) =
                if isaT value then
                    result <- Serilog.Events.ScalarValue(value)
                    true
                else
                    false
    type DontDestructureEvents = DontDestructureAssignableTo<Evt>
    let withEvent (value : Evt) (log : ILogger) =
        // This is the best way I've found to prevent the events from being destructured:
        // 1) request destructuring here
        // 2) configure the logger with `.Destructure.With<Log.DontDestructureEvents>()`
        // passing destructureObjects=false (which is the default), still subjects the context value to some destructuring
        let hack = true
        log.ForContext("esEvt", value, destructureObjects = hack)
```

This is less than ideal - perhaps if I could change the logging to be `log.ForContext("esEvt",Serilog.Events.ScalarValue(value),destructureObjects=false)` and have it just work without the config? Would a PR to that effect be accepted, or is there a better way already ?

**Comment by nblumhardt** [comment]
Hey, thanks for the follow-up! Hope the snow stayed frozen up there :-)

Here's one way to make it work, today, in a local fashion:

```csharp
class ScalarValueEnricher : ILogEventEnricher
{
    readonly LogEventProperty _prop;

    public ScalarValueEnricher(string name, object value)
    {
        _prop = new LogEventProperty(name, new ScalarValue(value));
    }

    public void Enrich(LogEvent evt, ILogEventPropertyFactory _)
    {
         evt.AddPropertyIfAbsent(_prop);
    }
}
```

Usage is:

```csharp
using (LogContext.Push(new ScalarValueEnricher("esEvt", value))
{
    Log.Information("This has a scalar attached..");
}
```

**Comment by bartelink** [comment]
Thanks, I love it; I used it like so:
```
 let withEvent (value : Log.Event) (log : ILogger) =
     let enrich (e : LogEvent) = e.AddPropertyIfAbsent(LogEventProperty("esEvt", ScalarValue(value)))
     log.ForContext({ new Serilog.Core.ILogEventEnricher with member __.Enrich(evt,_) = enrich evt })
```
... and I've a grin almost as wide as during yesterday's tobogganing ;)

**Comment by bartelink** [comment]
For others looking on: I also used this as a workaround to produce a clean render of something which makes sense as a dictionary without having to introduce an extra type with a Dictionary`2 embedded [here](https://github.com/jet/CallPolly/blob/c8f18726567be2addc5eb9875fb2c6050e2c3611/src/CallPolly/Context.fs#L45)
